include/ck/utility/get_id.hpp

```diff
@@ -1,12 +1,44 @@
 // SPDX-License-Identifier: MIT
-// Copyright (c) 2018-2023, Advanced Micro Devices, Inc. All rights reserved.
+// Copyright (c) 2018-2025, Advanced Micro Devices, Inc. All rights reserved.
 
 #pragma once
 
 #include "ck/ck.hpp"
 
 namespace ck {
 
+#if defined(CK_ENABLE_DYNAMIC_WARP_SIZE)
+__device__ constexpr index_t get_warp_size()
+{
+#if defined(__HIP_DEVICE_COMPILE__)
+#if defined(__GFX9__)
+    return 64;
+#else
+    return 32;
+#endif
+#else
+    return 64;
+#endif
+}
+
+inline __host__ index_t get_warp_size()
+{
+#if !(defined(__HIPCC_RTC__) || defined(CK_CODE_GEN_RTC))
+    hipDeviceProp_t props{};
+    int device;
+    auto status = hipGetDevice(&device);
+    if(status == hipSuccess)
+    {
+        status = hipGetDeviceProperties(&props, device);
+        if(status == hipSuccess)
+        {
+            return props.major > 9 ? 32 : 64;
+        }
+    }
+#endif
+    return 64;
+}
```

Comment:
FYI I previously found out that calling hipGetDeviceProperties is pretty expensive and can cause a lot of slowdown. I suggest to replace it with `hipDeviceGetAttribute` and/or cache the the device's warp size. Also, this should probably accept a `hipStream` in case the current device is not the one that's associated with the stream that the kernel should launch on. rocPRIM implementation: https://github.com/ROCm/rocm-libraries/blob/b10a470ec0c83944051e8ed53e632c360ee02e3b/projects/rocprim/rocprim/include/rocprim/device/config_types.hpp#L353-L396